### PR TITLE
[Suricata] Move edge processing to ingest pipelines

### LINKED ISSUE #662: Convert Suricata's edge processing to Ingest Node pipeline
# Convert edge processing to Ingest Node pipeline

This package uses Beats processors for some of its data processing. We want to
move that processing into the ingest node pipeline that is part of the package
to make reuse easier (e.g. data from Kafka could be routed through the pipeline).

### Data Streams
- eve (inputs: log)
  - [ ] All beats processors are removed
  - [ ] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) added

### PR DESCRIPTION
**Pull request by P1llus**
## What does this PR do?

This PR moves edge processing to ingest pipelines for Suricata.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.

## Related issues

- Closes #662

### CODE REVIEW

**Comment by P1llus** [comment]
This PR removes most of the edge processing, currently awaiting https://github.com/elastic/elasticsearch/pull/67611 to remove the last one.

Golden files has not been generated yet, will await until it is complete.

File: packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -5,7 +5,80 @@ processors:
...
+        - ISO8601
+  - community_id:
+      target_field: network.community_id
+      ignore_failure: true
+      if: ctx?.source?.port != null && ctx?.destination?.port != null
+  - registered_domain:
+      field: suricata.eve.dns.rrname
+      target_field: dns.question
+      ignore_missing: true
+      if: 'ctx?.suricata?.eve?.dns?.type == "query" || ctx?.suricata?.eve?.dns?.version == 2'
```
Comment (andrewkroh) [inline packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml:81]: Would it make a difference if this condition was removed? Seems like with ignore_missing: true there isn't much of a downside to simplifying by dropping the condition.

File: packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -5,7 +5,80 @@ processors:
...
+      ignore_missing: true
+      if: ctx?.event?.created != null
+  - date:
+      field: suricata.eve.timestamp
+      formats:
+        - ISO8601
+  - community_id:
+      target_field: network.community_id
+      ignore_failure: true
+      if: ctx?.source?.port != null && ctx?.destination?.port != null
```
Comment (andrewkroh) [inline packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml:76]: I think this condition could be removed. I assume that the processor will just not add any community ID when they are missing.

**Review by andrewkroh — APPROVED** [review]

File: packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (P1llus) [reply packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml:81]: I was a bit unsure, the conditions existed in the initial beat processor, so I thought maybe there would be other values there as well.

File: packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (P1llus) [reply packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml:76]: Let me run some tests, I have seen this failing sometimes if it is missing some mandatory ports, but that might not be the case.

**Pull request merged**